Mimir.Worker/BlockPoller.cs

```diff
@@ -60,25 +72,104 @@ public async Task RunAsync(CancellationToken cancellationToken)
             }
 
             var agentAddress = new Address(tx.Signer);
-            var avatarAddresses = Enumerable.Range(0, GameConfig.SlotCount)
+            var avatarAddresses = Enumerable
+                .Range(0, GameConfig.SlotCount)
                 .Select(e => Addresses.GetAvatarAddress(agentAddress, e));
-            var avatarDataArray = await Task.WhenAll(avatarAddresses.Select(stateGetter.GetAvatarData));
+            var avatarDataArray = await Task.WhenAll(
+                avatarAddresses.Select(stateGetter.GetAvatarData)
+            );
             await mongoDbStore.BulkUpsertAvatarDataAsync(
-                avatarDataArray
-                    .Where(e => e is not null)
-                    .OfType<AvatarData>()
-                    .ToList());
+                avatarDataArray.Where(e => e is not null).OfType<AvatarData>().ToList()
+            );
+        }
+    }
+
+    private async Task EveryPatchTableAsync(
+        long processBlockIndex,
+        CancellationToken cancellationToken
+    )
+    {
+        var rawPatchTableTxsResp = await headlessGqlClient.GetPatchTableTransactions.ExecuteAsync(
+            processBlockIndex,
+            cancellationToken
+        );
+        if (rawPatchTableTxsResp.Data is null)
+        {
+            HandleErrors(rawPatchTableTxsResp);
+            return;
+        }
+
+        var patchTableTxs = rawPatchTableTxsResp
+            .Data?.Transaction?.NcTransactions.Where(raw => raw is not null)
+            .Select(raw =>
+                TxMarshaler.DeserializeTransactionWithoutVerification(
+                    Convert.FromBase64String(raw!.SerializedPayload)
+                )
+            )
+            .ToList();
+        foreach (var patchTableTx in patchTableTxs)
+        {
+            var patchTableAction = (Dictionary)patchTableTx.Actions[0];
+            var patchTableActionValues = (Dictionary)patchTableAction["values"];
+            var tableName = ((Text)patchTableActionValues["table_name"]).ToDotnetString();
+
+            var sheetType = typeof(ISheet)
+                .Assembly.GetTypes()
+                .Where(type =>
+                    type.Namespace is { } @namespace
+                    && @namespace.StartsWith($"{nameof(Nekoyume)}.{nameof(Nekoyume.TableData)}")
+                    && !type.IsAbstract
+                    && typeof(ISheet).IsAssignableFrom(type)
+                    && type.Name == tableName
+                )
+                .FirstOrDefault();
```

Comment:
It would be better to declare `sheetTypes` outside of the `foreach` statement and reuse it.
```
var sheetTypes = typeof(ISheet)
                .Assembly.GetTypes()
                .Where(type =>
                    type.Namespace is { } @namespace
                    && @namespace.StartsWith($"{nameof(Nekoyume)}.{nameof(Nekoyume.TableData)}")
                    && !type.IsAbstract
                    && typeof(ISheet).IsAssignableFrom(type));
```

Reply:
thanks!